ui/components/app/wallet-overview/coin-overview.tsx

```diff
@@ -136,29 +241,93 @@ export const CoinOverview = ({
         >
           <div className={`${classPrefix}-overview__balance`}>
             <div className={`${classPrefix}-overview__primary-container`}>
-              {balance ? (
-                <UserPreferencedCurrencyDisplay
-                  style={{ display: 'contents' }}
-                  className={classnames(
-                    `${classPrefix}-overview__primary-balance`,
-                    {
-                      [`${classPrefix}-overview__cached-balance`]:
-                        balanceIsCached,
-                    },
-                  )}
-                  data-testid={`${classPrefix}-overview__primary-currency`}
-                  value={balance}
-                  type={
-                    showPrimaryCurrency(
-                      isOriginalNativeSymbol,
-                      useNativeCurrencyAsPrimaryCurrency,
-                    )
-                      ? PRIMARY
-                      : SECONDARY
-                  }
-                  ethNumberOfDecimals={4}
-                  hideTitle
-                />
+              {balanceToDisplay ? (
```

Comment:
I wonder if we should expand this condition to account for the `loading` boolean that gets exported from the `useAccountTotalFiatBalance` hook? This helps to prevent the visual jump between non-aggregated -> aggregated balances as balances get updated.

```suggestion
              {balanceToDisplay && !loading ? (
```

We still get some visual jump between the spinner and the balance amount, so I'm actually not sure which is preferable from a UI perspective. Should we ask? I think a skeleton loader of some sort would be cool, but that sounds out of scope for this task.

Reply:
That makes sense, Although i didn't think it is needed because i haven't noticed any difference visually when adding the `!loading`;

If the balanceToDisplay is undefined it will fallback to 
` <Spinner className="loading-overlay__spinner" />`

Reply:
Hmm, this is the difference I'm seeing:

Without `!loading`, it briefly flashes the nativeTokenBalance, before updating to aggregated balance after tokenBalances are fetched:
https://github.com/user-attachments/assets/6aeb9616-c4d3-4e69-92bb-295dfcdcd679

With `!loading` it briefly flashes the loading indicator, before updating to aggregated balance:
https://github.com/user-attachments/assets/c92aba5b-96e9-402a-93db-8cf1ae50b9af

Again, I'm not 100% sure what the intended UX is here. To me it's a bit weird to see a smaller balance show up before updating to the "final" version. But this doesn't need to block if it's intended.

Reply:
Thanks for the videos! 🙏 
Pushed here [e3eeb8a](https://github.com/MetaMask/metamask-extension/pull/27097/commits/e3eeb8ad7d2f2ca87c263ac5f41c63fb0a0449bb)

Reply:
Hello @MetaMask/wallet-ux 👋  using the hook `useAccountTotalFiatBalance`; the `totalFiatBalance` was resolving slightly after the native balance, which in the original code resulted, in seeing the native fiat balance for a split of second, then seing the total fiat balance. We added the `!loading` to avoid this behavior and show the loader until the total fiat balance resolves.

Reply:
I agree with @gambinish proposal. I think it would look better if there was a skeleton loader that was shown instead of it flashing. What do you guys think? @hesterbruikman @yanrong-chen?

This is probably out of scope anyhow and its a non-blocker but I think it would be nice